src/checkout/checkout-service.ts

```diff
@@ -1128,6 +1128,25 @@ export default class CheckoutService {
         return this._dispatch(action);
     }
 
+    /**
+     * Executes the spam protection for storefront
+     *
+     * If enabled, this is required for storefront protected actions
+     * like customer account creation.
+     *
+     * ```js
+     * await service.executeStorefrontSpamProtection({ containerId: 'recatpcha' });
+     * ```
+     *
+     * @param options - Options for initializing spam protection.
+     * @returns A promise that resolves to the current state.
+     */
+    executeStorefrontSpamProtection(options: SpamProtectionOptions): Promise<CheckoutSelectors> {
+        const action = this._spamProtectionActionCreator.initializeStorefront(options);
```

Comment:
I think if we go down the route of invisible reCaptcha, we probably don't need a separate method to initialise reCaptcha. The initialisation and verification could just be done as a part of `createCustomerAccount` call. You also don't need to store it in `CheckoutStore` because you can just pass the token along in the promise chain.